ast/location.go

```diff
@@ -126,6 +136,8 @@ func (sp *SourceProvider) GetSnippet(loc LocationRange) string {
 	return result.String()
 }
 
+// BuildSource transforms a source file string into a Source struct.
+// TODO: This seems like a job for strings.Split() with a final \n touch-up.
```

Comment:
Ditto

Reply:
I think you're right. Feel free to do it in this change or a separate one later, whatever is more convenient.